### DISCUSSION TITLE
Curso: Acesso à dados / Connection String, funciona no Visual Studio. No VS Code não.

### DISCUSSION THREAD

**Discussion by rogerfsp** [balta-io/forum#413] [Tira Dúvidas] [answered]
Olá,
No Visual Studio 2022, instalei o NuGet "System.Data.SqlClient" e, consigo me conectar com o banco balta normalmente, usando a seguinte string connection:

  "Server=localhost,1434;Database=balta;User Id=sa;Password=*******;"

No VS Code, instalei o NuGet  Microsoft.Data.SqlClient, e usando a mesma string connection, recebo o erro:

Unhandled exception. Microsoft.Data.SqlClient.SqlException (0x80131904): A connection was successfully established with the server, but then an error occurred during the login process. (provider: SSL Provider, error: 0 - A cadeia de certificação foi emitida por uma autoridade que não é de confiança.)

Ou seja, o erro diz que "a conexão foi estabelecida com sucesso", mas o restante da mensagem informa outros problemas.
Por favor, conto com o suporte de vocês em relação a isto.

Muito obrigado!

**Comment by rogerfsp** [comment, marked as answer]
Respondendo a mim mesmo:

Instalei outro pacote  (dotnet add package System.Data.SqlClient) e funcionou.
Por favor, apesar de eu ter conseguido usando outro pacote, comentários sobre o problema que ocorreu com o Microsoft.Data.SqlClient, serão muito bem vindos.

Muito obrigado.

**Reply by andrebaltieri** [reply]
Obrigado por compartilhar!!!

**Comment by ddcsilva** [comment]
Roger, experimente alterar seu método OnConfiguring do seu contexto para:

```
protected override void OnConfiguring(DbContextOptionsBuilder options)
        => options.UseSqlServer("Server=localhost,1433;Database=Blog;User ID=sa;Password=SENHA_AQUI;TrustServerCertificate=True");
```

Para fins de desenvolvimento local, você pode instruir o SQL Server a ignorar a cadeia de validação do certificado SSL, para isso basta adicionar `TrustServerCertificate=True` no final da string.
